RuntimeDependencyChecker blocks startup

# Bug Report

## Summary

Testing on Kubuntu (KDE Plasma) with the latest AppImage and Chrome, logging in is not detected as authorized.

## Steps to reproduce

Try it on KDE with Chrome.

## What is the expected behaviour?

For the app to detect it's logged in.

## Other information

[nexusmods.app.main.current.log](https://github.com/user-attachments/files/19291562/nexusmods.app.main.current.log)
[nexusmods.app.slim.00.log](https://github.com/user-attachments/files/19291564/nexusmods.app.slim.00.log)
[nexusmods.app.slim.01.log](https://github.com/user-attachments/files/19291561/nexusmods.app.slim.01.log)
[nexusmods.app.slim.02.log](https://github.com/user-attachments/files/19291560/nexusmods.app.slim.02.log)
[nexusmods.app.slim.current.log](https://github.com/user-attachments/files/19291563/nexusmods.app.slim.current.log)

![Image](https://github.com/user-attachments/assets/97ced8b0-b147-47de-ae02-b891ace8c12e)

Oh god, it stalled for 4 minutes!!

CC. @erri120 

```
00:00:00.353 [ERROR] (NexusMods.DataModel.GameRegistry.GameRegistry) Exception in locator|System.Text.Json.JsonException: The JSON value could not be converted to GameFinder.Launcher.Heroic.DTOs.Installed. Path: $.installed[0].buildId | LineNumber: 13 | BytePositionInLine: 22.
 ---> System.InvalidOperationException: Cannot get the value of a token type 'Number' as a string.
   at System.Text.Json.ThrowHelper.ThrowInvalidOperationException_ExpectedString(JsonTokenType tokenType)
   at System.Text.Json.Utf8JsonReader.GetString()
   at System.Text.Json.Serialization.JsonConverter`1.TryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value, Boolean& isPopulatedValue)
   at System.Text.Json.Serialization.JsonConverter`1.TryReadAsObject(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, Object& value)
   at System.Text.Json.Serialization.Converters.LargeObjectWithParameterizedConstructorConverter`1.ReadAndCacheConstructorArgument(ReadStack& state, Utf8JsonReader& reader, JsonParameterInfo jsonParameterInfo)
   at System.Text.Json.Serialization.Converters.ObjectWithParameterizedConstructorConverter`1.ReadConstructorArgumentsWithContinuation(ReadStack& state, Utf8JsonReader& reader, JsonSerializerOptions options)
   at System.Text.Json.Serialization.Converters.ObjectWithParameterizedConstructorConverter`1.OnTryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value)
   at System.Text.Json.Serialization.JsonConverter`1.TryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value, Boolean& isPopulatedValue)
   at System.Text.Json.Serialization.JsonCollectionConverter`2.OnTryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, TCollection& value)
   at System.Text.Json.Serialization.JsonConverter`1.TryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value, Boolean& isPopulatedValue)
   at System.Text.Json.Serialization.Converters.SmallObjectWithParameterizedConstructorConverter`5.TryRead[TArg](ReadStack& state, Utf8JsonReader& reader, JsonParameterInfo jsonParameterInfo, TArg& arg)
   at System.Text.Json.Serialization.Converters.SmallObjectWithParameterizedConstructorConverter`5.ReadAndCacheConstructorArgument(ReadStack& state, Utf8JsonReader& reader, JsonParameterInfo jsonParameterInfo)
   at System.Text.Json.Serialization.Converters.ObjectWithParameterizedConstructorConverter`1.ReadConstructorArgumentsWithContinuation(ReadStack& state, Utf8JsonReader& reader, JsonSerializerOptions options)
   at System.Text.Json.Serialization.Converters.ObjectWithParameterizedConstructorConverter`1.OnTryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value)
   at System.Text.Json.Serialization.JsonConverter`1.TryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value, Boolean& isPopulatedValue)
   at System.Text.Json.Serialization.JsonConverter`1.ReadCore(Utf8JsonReader& reader, T& value, JsonSerializerOptions options, ReadStack& state)
   --- End of inner exception stack trace ---
   at System.Text.Json.ThrowHelper.ReThrowWithPath(ReadStack& state, Utf8JsonReader& reader, Exception ex)
   at System.Text.Json.Serialization.JsonConverter`1.ReadCore(Utf8JsonReader& reader, T& value, JsonSerializerOptions options, ReadStack& state)
   at System.Text.Json.Serialization.Metadata.JsonTypeInfo`1.ContinueDeserialize(ReadBufferState& bufferState, JsonReaderState& jsonReaderState, ReadStack& readStack, T& value)
   at System.Text.Json.Serialization.Metadata.JsonTypeInfo`1.Deserialize(Stream utf8Json)
   at GameFinder.Launcher.Heroic.HeroicGOGHandler.ParseInstalledJsonFile(AbsolutePath path, AbsolutePath configPath)+MoveNext()
   at GameFinder.Launcher.Heroic.HeroicGOGHandler.FindAllGames()+MoveNext()
   at System.Linq.Enumerable.<ToArray>g__EnumerableToArray|314_0[TSource](IEnumerable`1 source)
   at GameFinder.Common.Extensions.SplitResults[TGame](IEnumerable`1 results)
   at GameFinder.Common.AHandler`2.FindAllGamesById(ErrorMessage[]& errors)
   at NexusMods.StandardGameLocators.HeroicGogLocator.Find(ILocatableGame game)+MoveNext() in /_/src/NexusMods.StandardGameLocators/HeroicGogLocator.cs:line 37
   at NexusMods.DataModel.GameRegistry.GameRegistry.FindInstallations()+MoveNext() in /_/src/NexusMods.DataModel/GameRegistry/GameRegistry.cs:line 157
```

A Heroic related parse error, I thought I'd bring it to your attention.

--- a/src/NexusMods.CrossPlatform/Process/RuntimeDependencyChecker.cs
+++ b/src/NexusMods.CrossPlatform/Process/RuntimeDependencyChecker.cs
@@ -6,7 +6,7 @@ using Microsoft.Extensions.Logging;
 namespace NexusMods.CrossPlatform.Process;
 
 [UsedImplicitly]
-internal class RuntimeDependencyChecker : IHostedService
+internal class RuntimeDependencyChecker : BackgroundService
 {
     private readonly IRuntimeDependency[] _runtimeDependencies;
     private readonly ILogger _logger;
@@ -17,28 +17,27 @@ internal class RuntimeDependencyChecker : IHostedService
         _logger = serviceProvider.GetRequiredService<ILogger<RuntimeDependencyChecker>>();
     }
 
-    public async Task StartAsync(CancellationToken cancellationToken)
+    protected override Task ExecuteAsync(CancellationToken stoppingToken)
     {
-        foreach (var dep in _runtimeDependencies)
+        return Task.Run(async () =>
         {
-            cancellationToken.ThrowIfCancellationRequested();
-            try
+            foreach (var dep in _runtimeDependencies)
             {
-                var information = await dep.QueryInstallationInformation(cancellationToken);
-                if (information.HasValue)
-                {
-                    _logger.LogInformation("{Name}: {Information}", dep.DisplayName, information.Value);
-                } else {
-                    _logger.LogWarning("{Name}: couldn't query information", dep.DisplayName);
-                }
-            } catch (Exception e) {
-                _logger.LogError(e, "{Name}: exception while querying information", dep.DisplayName);
-            }
-        }
-    }
+                if (stoppingToken.IsCancellationRequested) return;
 
-    public Task StopAsync(CancellationToken cancellationToken)
-    {
-        return Task.CompletedTask;
+                try
+                {
+                    var information = await dep.QueryInstallationInformation(stoppingToken).ConfigureAwait(false);
+                    if (information.HasValue)
+                    {
+                        _logger.LogInformation("{Name}: {Information}", dep.DisplayName, information.Value);
+                    } else {
+                        _logger.LogWarning("{Name}: couldn't query information", dep.DisplayName);
+                    }
+                } catch (Exception e) {
+                    _logger.LogError(e, "{Name}: exception while querying information", dep.DisplayName);
+                }
+            }
+        }, stoppingToken);
     }
 }

